contrib/orafce/charpad.c

```diff
@@ -57,17 +58,49 @@ orafce_lpad(PG_FUNCTION_ARGS)
 	bool	s2_operate = ON,
 			half_space = OFF,
 			init_ptr = ON;
+	int		nargs;
+	Oid		argoid1,
+			argoid2,
+			argoid3;
+
+	nargs = PG_NARGS();
+	if (nargs < 2)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				errmsg("too few args.")));
+	if (nargs > 3)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				errmsg("more args than function needed.")));
+
+	argoid1 = get_fn_expr_argtype(fcinfo->flinfo, 0);
+
+	if (BPCHAROID == argoid1 || TEXTOID == argoid1)
+		string1 = PG_GETARG_TEXT_P(0);
+	else
+		string1 = (text *)orafce_sourcetype_to_targetype(PG_GETARG_DATUM(0), argoid1, TEXTOID);
+	/* get byte-length of the 1st and 3rd argument strings */
+	s1blen = VARSIZE_ANY_EXHDR(string1);
+
+	argoid2 = get_fn_expr_argtype(fcinfo->flinfo, 1);
+	output_width = (int32)DatumGetFloat8(orafce_sourcetype_to_targetype(PG_GETARG_DATUM(1), argoid2, FLOAT8OID));
+
+	if (nargs == 3)
+	{
+		argoid3 = get_fn_expr_argtype(fcinfo->flinfo, 2);
+		string2 =  (text *)orafce_sourcetype_to_targetype(PG_GETARG_DATUM(2), argoid3, TEXTOID);
+	}
+	else
+		string2 = cstring_to_text(" ");
+
+	s2blen = VARSIZE_ANY_EXHDR(string2);
```

Comment:
I think it's better to initialise 'string2' at declaration and have 's2blen' conditional instead of assigning a empty string just for this.

```suggestion
string2 = NULL;
...
if (string2)
	s2blen = VARSIZE_ANY_EXHDR(string2);
```